language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java

```diff
@@ -91,30 +98,61 @@ public List<CodeAction> getCodeActions(CodeActionContext context,
             return Collections.emptyList();
         }
         
-        String constName = getLocalVarName(context);
-        String value = node.toSourceCode().strip();
-        LineRange replaceRange = node.lineRange();
+        String constName = getConstantName(context);
         Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get().typeOf(node);
         if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
             return Collections.emptyList();
         }
-        
         Position constDeclPosition = getPosition(context);
-        String constDeclStr = String.format("const %s %s = %s;%n", typeSymbol.get().signature(), constName, value);
-        TextEdit constDeclEdit = new TextEdit(new Range(constDeclPosition, constDeclPosition), constDeclStr);
-        TextEdit replaceEdit = new TextEdit(new Range(PositionUtil.toPosition(replaceRange.startLine()),
-                PositionUtil.toPosition(replaceRange.endLine())),  constName);
+        
+        // Check if the selection is a range or a position, and whether quick picks are supported by the client
+        LSClientCapabilities lsClientCapabilities = context.languageServercontext().get(LSClientCapabilities.class);
+        if (isRange(context.range()) || !lsClientCapabilities.getInitializationOptions().isQuickPickSupported()) {
+            
+            // Selection is a range
+            List<TextEdit> textEdits = getTextEdits(node, typeSymbol.get(), constName, constDeclPosition);
+            return Collections.singletonList(CodeActionUtil.createCodeAction(CommandConstants.EXTRACT_TO_CONSTANT,
+                    textEdits, context.fileUri(), CodeActionKind.RefactorExtract));
+        }
+        
+        // Selection is a position
+        List<Node> nodeList = new ArrayList<>();
+        // Identify the sub-expressions to be extracted
+        while (!(node instanceof StatementNode) && !(node instanceof ModuleMemberDeclarationNode)
```

Comment:
Shall we move this validation to a separate private method.  Something like `isExpressionNode` ?